features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Details/Overview.jsx

```diff
@@ -165,6 +165,89 @@ function Overview() {
+                id: 'Apis.Details.Overview.select.env.error',
+                defaultMessage: 'Error Selecting Environment',
+            }));
+        }
+    };
+
+    const getSubscriptionPolicies = async () => {
+        const restApi = new API();
+        return restApi.getAllTiers('subscription')
+            .then((response) => {
+                try {
+                // Filter policies base on async or not.
+                    const filteredList = response.body.list.filter((str) => isApiPolicy(str.name));
+                    setAllPolicies(filteredList);
+                } catch (e) {
+                    console.log(e);
+                    Alert.error(intl.formatMessage({
+                        id: 'Apis.Details.Overview.error.occurred',
+                        defaultMessage: 'Error occurred',
+                    }));
+                }
+            }).catch((error) => {
+                if (process.env.NODE_ENV !== 'production') {
+                    console.log(error);
+                }
+                const { status } = error;
+                if (status === 404) {
+                    Alert.error(intl.formatMessage({
+                        id: 'Apis.Details.Overview.error.occurred',
+                        defaultMessage: 'Error occurred',
+                    }));
+                    setNotFound(true);
+                }
+                setAllDocuments([]);
+            });
+    };
+
+    const getDocuments = async () => {
+        const restApi = new API();
+        return restApi.getDocumentsByAPIId(api.id)
+            .then((response) => {
+                const overviewDoc = response.body.list.filter((item) => item.otherTypeName === '_overview');
+                if (overviewDoc.length > 0) {
+                // We can override the UI with this content
+                    setOverviewDocOverride(overviewDoc[0]); // Only one doc we can render
+                }
+                setAllDocuments(response.body.list);
+            })
+            .catch((error) => {
+                if (process.env.NODE_ENV !== 'production') {
+                    console.log(error);
+                }
+                const { status } = error;
+                if (status === 404) {
+                    Alert.error(intl.formatMessage({
+                        id: 'Apis.Details.Overview.error.occurred',
+                        defaultMessage: 'Error occurred',
+                    }));
+                    setNotFound(true);
+                }
+                setAllDocuments([]);
+            });
+    };
+    useEffect(() => {
+        setIsLoading(true);
+        const { endpointURLs } = api;
+        if (endpointURLs && endpointURLs.length > 0) {
+            setSelectedEndpoint(endpointURLs[0]);
+        }
+        Promise.all([getDocuments(), getSubscriptionPolicies()])
+            .then(() => {
+                setIsLoading(false);
+            }).catch(() => {
+                setIsLoading(false);
```

Comment:
🤯  we need to log the error and show an error message to the user, We have swallowed the exception here , This will no clue if any exception occurred

Reply:
since the exception is logged in the method, we do not need to log it here right ?

Reply:
change. the logic here, issue was i was trying to catch an already handle scenario, therefore removing this catch clause and setting the loading to false inside the method would handle this properly https://github.com/wso2/carbon-apimgt/pull/10298/commits/fd68d8dfc8f1759263ac96dd2593dbd1842b1bdb#diff-6cbc7640b466c4fa1e26f18a3ab1692bb1c0c01331b628afa9a9faa4dcad0b78R233